src/Umbraco.AuthorizedServices/AuthorizedServicesComposer.cs

```diff
@@ -16,8 +15,18 @@ internal class AuthorizedServicesComposer : IComposer
 {
     public void Compose(IUmbracoBuilder builder)
     {
-        IConfigurationSection configSection = builder.Config.GetSection("Umbraco:AuthorizedServices");
-        builder.Services.Configure<AuthorizedServiceSettings>(configSection);
+        builder.Services.AddOptions<AuthorizedServiceSettings>()
+            .BindConfiguration("Umbraco:AuthorizedServices")
+            .Configure(options =>
+            {
+                // Automatically set the alias based on the key
+                foreach (var service in options.Services)
+                {
+                    service.Value.Alias = service.Key;
```

Comment:
You could use the dictionary key to get the alias from an `AuthorizedServiceSettings` instance and getting the `ServiceDetail` named options instance using `IOptionsMonitor<ServiceDetail>("serviceAlias")` requires you to already know the alias, but rewriting all usages of the `Alias` property would require a larger refactor.

Instead, I've opted to automatically set this property when configuring the instance.